static/js/app.ide.js

```diff
@@ -392,31 +413,85 @@ export default class IDEApp extends App {
    * @returns {Promise<void>}
    */
   async openLFSFolder() {
-    let rootFolderHandle;
     try {
-      rootFolderHandle = await window.showDirectoryPicker();
-    } catch (err) {
-      // User most likely aborted.
-      return;
-    }
+      let rootFolderHandle = await this._openLFS();
 
-    const hasPermission = await this.vfs._verifyLFSHandlePermission(rootFolderHandle);
-    if (hasPermission) {
-      // Make sure GitFS is stopped.
+      // Make sure GitFS is stopped before connecting LFS,
+      // and VFS cache for Git is cleared.
       if (this.hasGitFSWorker()) {
+        await this.vfs.clear();
         this.gitfs.terminate();
         this.gitfs = null;
       }
 
       fileTreeManager.removeLocalStorageWarning();
-      localStorageManager.setLocalStorageItem('use-lfs', true);
       this.closeAllFiles();
-      await idbManager.saveHandle('lfs', 'root', rootFolderHandle);
       fileTreeManager.setTitle(rootFolderHandle.name);
       pluginManager.triggerEvent('onStorageChange', 'lfs');
 
+      await this.vfs.connect(rootFolderHandle);
+
       // Render the LFS contents.
       await fileTreeManager.createFileTree();
     }
+    catch { }
+  }
+
+  async _reopenLFS() {
```

Comment:
@kkoomen Ik wil deze functie en de andere eronder verplaatsen naar de idb.js.
Die idb class krijg dan de rol om alles met LFS root handles te regelen, want dat is een hoop technische dingetjes die de app.ide.js complex maken. 
Die kan dan misschien beter lfs.js heten.
Heb je een mening hierover? (en nee, het heeft niet héél veel met de vfs worker te maken)

Reply:
@stgm Ja dat lijkt me prima! En inderdaad vfs-client.js -> vfs.js zou ook fijn zijn :) 

Ik zou `idb.js` niet in de fs folder doen, omdat het niet een fs zelf is of een core onderdeel, zoals de workers.

Reply:
Denk dat idb.js echt nergens anders voor gebruikt gaat worden, maar voor nu even achtergelaten in de code root.